Ramp3.0 single name for ramp fast cata

KeithKelleher opened the pull request:
also, there's a new database
and you can load databases from non-main branches when you pass an argument for the branch

for example:
db = RaMP(branch="ramp3.0")

Mathelab reviewed (commented):
Have you tested the new vignette here https://github.com/ncats/RaMP-DB/blob/gh-pages/Updated_RaMP_Vignette.Rmd?

KeithKelleher commented:
:) I added another commit after testing the vignette. 
there were two fixes

- we removed an output column from rampFastCata because the data wasn't used anywhere, but the column name was still used as a check for plotCataNetwork to know the input was formatted right. I updated the check to not check for it because it doesn't use it anyway.
- I made a fix to chemicalClassSurvey and chemicalClassEnrichment because they weren't returning anything. Here is the relevant information that I posted on slack about this one.

  - chemicalClassSurvey and chemicalClassEnrichment don’t return anything
    - chemical.classes <- chemicalClassSurvey(mets = metabolites.of.interest, db=rampDB)
    - but it doesn’t using the old version (2.5.4) either
    - here’s the problem: it is fetching data from the database, but it’s all getting filtered away with these checks
  ```
  \# need to filter for our specific source ids
  if(inferIdMapping) {
    metsData2 <- subset(metsData, “sourceId” %in% mets)
    metsData <- metsData2
  } else {
    metsData <- subset(metsData, “class_source_id” %in% mets)
  }
  ```
    - Having looked into this issue…
      - It seems like these two lines were updated recently, to include quotes around the column names
        -   `metsData2 <- subset(metsData, “sourceId” %in% mets)`
        -   `metsData <- subset(metsData, “class_source_id” %in% mets)`
      - b/c previously without those quotes, they refer to variables which don’t exist
        - `metsData <- subset(metsData, sourceId %in% mets)`
        - `metsData <- subset(metsData, class_source_id %in% mets)`
      - so previously, that filtering ended up doing nothing, b/c sourceId and class_source_id don’t exist
      - and, fixing that code broke the function, because it was relying on that filtering to not do anything
    - So let’s delete those filtering steps from chemicalClassSurvey and chemicalClassEnrichment
      - if that filtering worked, it would limit you to inputting hmdb IDs and LIPIDMAPS IDs into those two functions, b/c those are the only two sources of chemical classes

KeithKelleher commented:
I reverted the change to the chemicalClass functions.
